pygfx/objects/_events.py

```diff
@@ -333,6 +338,32 @@ def dispatch_event(self, event: Event):
             event._retarget(captured_target)
             event.stop_propagation()
 
+        # Update the target tracker on all `pointer_move` events
+        if event.type == EventType.POINTER_MOVE:
+            # Check if this is the first move event for the pointer
+            first_move = pointer_id not in RootEventHandler.target_tracker
+            # Get the previous target for this pointer (if any)
+            previous_target_ref = RootEventHandler.target_tracker.get(pointer_id)
+            previous_target = (previous_target_ref and previous_target_ref()) or None
+            # Get the current target for this pointer (if any)
+            new_target = event.target
+            # Check if the target has changed since the previous move event
+            if previous_target is not new_target or first_move:
+                # Update the current target for this pointer
+                RootEventHandler.target_tracker[pointer_id] = (
+                    new_target and ref(new_target)
+                ) or None
+                if not first_move:
```

Comment:
IIUC both `previous_target`  and `new_target` are either an object or `None`. Can't we simply do something like this?
```py
if new_target is not previous_target:
     ... store new target in tracker
    if previous_target:
        ... dispatch LEAVE
    if new_target:
      ... dispatch ENTER
```

Reply:
The complication here is that `None` represents the event root (the renderer), and so, on the very first move event, the root needs to receive a pointer_enter event even though both previous and new target are `None`.

Reply:
See #296 for an alternative implementation where the root is also tracked